src/main/java/com/agricraft/agricore/json/AgriJsonVersion.java

```diff
@@ -0,0 +1,21 @@
+package com.agricraft.agricore.json;
+
+import javax.annotation.Nullable;
+import java.util.function.Function;
+
+public interface AgriJsonVersion<T extends AgriSerializable> {
```

Comment:
I should have added this to begin with.

Reply:
Well you did a great job with the initial implementation overall, and futureproofing something is quite hard...